rust/actyx/certs/src/bin/ax-cert.rs

```diff
@@ -30,15 +32,90 @@ struct AppLicenseOpts {
     #[structopt(long)]
     app_id: AppId,
 
-    /// ISO 8601 (i.e. 2014-11-28T12:00:09Z) expiration date time
+    /// An expiration date time in ISO 8601 (i.e. 2014-11-28T12:00:09Z).
+    /// Takes precedences over `--expires-in`.
     #[structopt(long)]
-    expires_at: DateTime<Utc>,
+    expires_at: Option<DateTime<Utc>>,
+
+    /// The amount of time in which the license should expire. The accepted format is
+    /// composed of a number followed by a unit (i.e. "10Y"), whitespace is supported
+    /// before, between and after the number and the unit. Values are accepted in
+    /// descending order according to the size of the unit, the unit character and
+    /// ordering is as follows:
+    ///
+    /// [Y]ear, [M]onth, [w]eek, [d]ay, [h]our, [d]ay, [m]inute, [s]econd
+    ///
+    /// Note: Years are considered to be 365 days long and months to be 30 days long.
+    ///
+    /// For example: "1Y 3M 4h" means that the license should expire in 1 year, 3 months and 4 hours.
+    #[structopt(long, parse(try_from_str = parse_expires_in))]
+    expires_in: Option<DateTime<Utc>>,
 
     /// Requester's email address
     #[structopt(long)]
     email: String,
 }
 
+/// Parsing function for the `expires_in` variable.
+/// Accepts strings matching the regex:
+/// `^\s*([0-9]+Y)?\s*([0-9]+Y)?\s*([0-9]+Y)?\s*([0-9]+Y)?\s*([0-9]+Y)?\s*([0-9]+Y)?\s*([0-9]+Y)?\s*$`
+///
+/// Y - year(s) - counted as 365 days
+/// M - month(s) - counted as 30 days
+/// w - week(s)
+/// d - day(s)
+/// h - hour(s)
+/// m - minute(s)
+/// s - second(s)
+fn parse_expires_in(expires_in: &str) -> Result<DateTime<Utc>, anyhow::Error> {
+    parse_expires_in_as_duration(expires_in).map(|d| Utc::now() + d)
+}
+
+// This function is much easier to test for correctness
+fn parse_expires_in_as_duration(expires_in: &str) -> Result<chrono::Duration, anyhow::Error> {
+    lazy_static! {
```

Comment:
curious, why does this need to be a lazy_static?

Reply:
Because only trivial (i.e. `const`) initializers are allowed for `static` names, so the regex needs to be parsed and compiled upon first use.

Reply:
ah i see. thanks for the explanation

Reply:
(I think, lmk if this is wrong) In this case the performance difference is irrelevant, we're matching the value once, so using `lazy_static` or just the `Regex::new` will be pretty much the same. I did it more out of an habit rather than "this is performance critical and I need this here"

Reply:
Understandable and I don't see why not.
Since, if the function is moved and reused eventually this will be a boon.